src/libraries/System.Console/src/System/IO/StdInReader.cs

```diff
@@ -367,7 +378,7 @@ internal bool MapBufferToConsoleKey(out ConsoleKey key, out char ch, out bool is
         /// not work, we simply return the char associated with that
         /// key with ConsoleKey set to default value.
         /// </summary>
-        public unsafe ConsoleKeyInfo ReadKey(out bool previouslyProcessed)
+        private unsafe ConsoleKeyInfo ReadKeyCore(bool convertCrToNl, out bool previouslyProcessed)
```

Comment:
This method can return cached keys.
Probably that means values pushed into this cache should be read under convertCrToNl: false, so we may/may not convert them when popping them here?

Reply:
That's a good point. I changed the code so that no conversions at all happen at the native layer. I've included logic for converting CR to LF characters for the `ReadLine()` implementation in the managed layer.